migration/pkg.go

```diff
@@ -0,0 +1,34 @@
+package migration
+
+import (
+	"github.com/konveyor/controller/pkg/logging"
+	"gorm.io/gorm"
+)
+
+var log = logging.WithName("migration")
+
+//
+// VersionKey is the setting containing the migration version.
+const VersionKey = ".migration.version"
+
+//
+// Version represents the value of the .migration.version setting.
+type Version struct {
+	Version int `json:"version"`
+}
+
+//
+// Migration encapsulates the functionality necessary to perform a migration.
+type Migration interface {
+	Apply(*gorm.DB) error
+	DisplayName() string
```

Comment:
One pedantic request:  Even though we discussed this as a "display name", I generally think it's better for fields/methods not be named for how they are used and instead just the what they are.
As such would recommend: `Name()`.